### ISSUE TITLE
[Feature]: Access and combine raw logits at inference time

### ISSUE THREAD

**Issue by WrtTZ** [vllm-project/vllm#18289]
### 🚀 The feature, motivation and pitch

We are working on decoding algorithms that combines the raw logits of two models and then run inference based on the combined logits.

More specifically, we'd like to load two models (potentially on different servers/processes/GPUs) using vllm, and then at every inference step, extract the raw logits of both models, combine them using a self-defined function and perform (greedy/beam-search) inference based on the combined logits.

### Alternatives

An alternative solution is to extend a logits processor to include another model. Model 1 will use this logit processor which takes in model 2 and prompt as arguments, run decoding to get the logits from Model 2 and combine them with Model 1.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by DarkLight1337** [comment]
See #12249 , #18176

**Comment by WrtTZ** [comment]
> See [#12249](https://github.com/vllm-project/vllm/issues/12249) , [#18176](https://github.com/vllm-project/vllm/issues/18176)

Thank you for the reference, do you have an estimated time of completion for [#12249](https://github.com/vllm-project/vllm/issues/12249)? Specifically, an attribute of the model called lm_head is mentioned which I believe is the embedding matrix to map the hidden states to raw logits. What's the current available endpoint to access that? Thank you!

**Comment by DarkLight1337** [comment]
We are planning to merge https://github.com/vllm-project/vllm/pull/18015 first before looking to integrate the common code into the existing model runner

**Comment by cuttle-fish-my** [comment]
Any progress now?
